silence/silence.go

```diff
@@ -638,11 +631,14 @@ func (s *Silences) Set(sil *pb.Silence) (string, error) {
 		sil.StartsAt = now
 	}
 
-	if err = s.setSilence(sil, now, false); err != nil {
-		return "", err
+	// If we got here it's either a new silence or a replacing one.
+	if s.limits.MaxSilences > 0 {
+		if len(s.st)+1 > s.limits.MaxSilences {
+			return sil.Id, fmt.Errorf("exceeded maximum number of silences: %d (limit: %d)", len(s.st), s.limits.MaxSilences)
```

Comment:
I return `sil.Id` here as that's what happens if `s.setSilence` returns an error too. `s.setSilence` can return an error if the silence is invalid or maximum size has been exceeded.